sync/src/main/java/tech/pegasys/artemis/sync/SyncManager.java

```diff
@@ -49,23 +54,41 @@ public SyncManager(
       return CompletableFuture.completedFuture(null);
     }
 
+    syncActive.set(true);
     Eth2Peer syncPeer = possibleSyncPeer.get();
     PeerSync peerSync = new PeerSync(syncPeer, storageClient, blockImporter);
 
     return peerSync
-        .sync()
-        .thenCompose(
-            result ->
-                result != PeerSyncResult.SUCCESSFUL_SYNC
-                    ? executeSync()
-                    : CompletableFuture.completedFuture(null));
+            .sync()
+            .thenCompose(
+                    result -> {
+                      if (result != PeerSyncResult.SUCCESSFUL_SYNC) {
+                        return executeSync();
+                      } else {
+                        syncActive.set(false);
+                        return CompletableFuture.completedFuture(null);
+                      }
+                    });
+
   }
 
   private Optional<Eth2Peer> findBestSyncPeer() {
-    UnsignedLong ourFinalizedEpoch = storageClient.getFinalizedEpoch();
     return network
         .streamPeers()
-        .filter(peer -> peer.getStatus().getFinalizedEpoch().compareTo(ourFinalizedEpoch) > 0)
+        .filter(this::isSyncSuitable)
         .max(Comparator.comparing(p -> p.getStatus().getFinalizedEpoch()));
   }
+
+  private void onNewPeer(Eth2Peer peer) {
+    if (isSyncSuitable(peer)) {
+      if (!syncActive.get()) {
+        executeSync();
+      }
+    }
+  }
+
+  private boolean isSyncSuitable(Eth2Peer peer) {
+    UnsignedLong ourFinalizedEpoch = storageClient.getFinalizedEpoch();
+    return peer.getStatus().getFinalizedEpoch().compareTo(ourFinalizedEpoch) > 0;
```

Comment:
I think currently when we receive a peer connected event, it won't yet have it's status.  We may need to change when the peer connected event fires or use `Eth2Peer.subscribeInitialStatus` in some way to delay considering the peer for syncing until it has a status.

Reply:
I think this is now handled in #1066